### PR TITLE
Spawner: Corrected interpretation of return values of waitpid

### PR DESCRIPTION
**Pull request by maltewi**

### CODE REVIEW

File: src/Spawner.cpp [hunk 1]
```diff
@@ -186,40 +186,47 @@ bool Spawner::ProcessHandle::alive() const
...
+    if(ret == -1 )
     {
         throw std::runtime_error(std::string("WaitPid failed ") + strerror(errno));
     }
-    
-    if(!status)
-    {
+    else if(ret == 0){
+        //Not yet terminated, but also no error.. so it's still alive
+        std::cout << "Process " << pid << " is still running" << std::endl;
```
Comment (saarnold) [inline src/Spawner.cpp:199]: I'm not sure how often `alive()` is called, but this might be better as a debug output.

**Comment by skasperski** [comment]
What is the purpose of `alive()` in this context? I think there should be different methods to check whether the process is alive and another to wait for it after sending SIGINT or SIGKILL.

File: src/Spawner.cpp (hunk 1 above)
Comment (maltewi) [reply src/Spawner.cpp:199]: It's possible that it gets called a few times before the process was terminated. If we had here a proper text-logging (eg base-logging) already in this package, I'd chosen debug print... so I agree with your comment.

But I'd prefer to consider integrating a text logging library as a separated issue. And in that go a categorization of all textlogs we have here in orocos_cpp (see #24).  @saarnold, please resolve conversation if you agree

**Comment by maltewi** [comment]
> What is the purpose of `alive()` in this context?

The function `alive` states if a previously started process is still running or not. It's used to check if starting of procss worked correctly, if termination requests are completed and to see if processes are still running okay.

> I think there should be different methods to check whether the process is alive and another to wait for it after sending SIGINT or SIGKILL.

There is the function for checking if a process is alive.. that's the alive function and I think it should stay as it is, it nicely does what it's name says.
There *could* be an additional convenience function being added that send a termination signal and then calls alive in a look until the process finished (such as killall() does). Currently it is expected that this loop and checking is implemented by the software that uses orocos_cpp (orogen-execution in our case), which is a likewise valid choice.

For the sake of keeping code changes minimal when providing bugfixes I propose to stay with the current scheme (software that sends the kill request has to check if it's completed).

**Pull request merged**